src/seismometer/report/fairness.py

```diff
@@ -0,0 +1,445 @@
+<td style="text-align: left;">{cls.CRITICAL_HIGH.value} more than {2*limit:.2%} greater than the largest cohort</td>
+</tr>
+<tr style="background: none;">
+<td style="text-align: left;">{cls.UNKNOWN.value} fewer than {censor_threshold} samples, data was censored</td>
+</tr>
+</details>"""
+        )
+
+    @classmethod
+    def get_fairness_icon(cls, ratio, limit: float = 0.2) -> "FairnessIcons":
+        """
+        Icon for fairness ratio
+        If fairness ratio is 0.2 (20%) we want to show a warning if we are outside this range and
+        a critical warning if we are 2x outside this range
+
+        4/5 - 5/4 is symmetric around 1
+
+        So we are looing at 1-ratio and 1/(1-ratio)
+
+        the limit is required to be strictly between 0 and 0.5.
+        """
+        lower_limit, upper_limit = 1 - limit, 1 / (1 - limit)
+        twice_lower_limit, twice_upper_limit = 1 - 2 * limit, 1 / (1 - 2 * limit)
+        if ratio is None or np.isnan(ratio):
+            return FairnessIcons.UNKNOWN
+        if ratio > twice_upper_limit:
+            return FairnessIcons.CRITICAL_HIGH
+        if ratio < twice_lower_limit:
+            return FairnessIcons.CRITICAL_LOW
+        if ratio > upper_limit:
+            return FairnessIcons.WARNING_HIGH
+        if ratio < lower_limit:
+            return FairnessIcons.WARNING_LOW
+        if ratio == 1:
+            return FairnessIcons.DEFAULT
+        return FairnessIcons.GOOD
+
+
+# endregion
+# region Fairness Table
+
+
+def sort_fairness_table(dataframe: pd.DataFrame, cohort_groups: tuple[str]):
+    """
+    Generates a sort key for the fairness table based on Cohort group name and Count
+
+    Parameters
+    ----------
+    dataframe : pd.DataFrame
+        Dataframe to sort
+    cohort_groups : tuple[str]
+        Cohort group names for sorting.
+    """
+
+    def fairness_sort_key(key: pd.Series) -> pd.Series:
+        match key.name:
+            case "Count":
+                return key.apply(lambda x: -x if x not in [FairnessIcons.UNKNOWN.value, "--"] else 0)
+            case "Cohort":
+                return key.apply(lambda x: cohort_groups.index(x))
+            case _:
+                return key
+
+    return dataframe.sort_values(by=["Cohort", "Count"], key=fairness_sort_key)
+
+
+def fairness_table(
+    dataframe: pd.DataFrame,
+    metric_fn: Callable[..., dict[str, float]],
+    metric_list: list[str],
+    fairness_ratio: float,
+    cohort_dict: dict[str, tuple[Any]],
+    *,
+    censor_threshold: int = 10,
+    **kwargs,
+) -> HTML:
+    fairness_data = pd.DataFrame()
+    metric_data = pd.DataFrame()
+    fairness_icons = pd.DataFrame()
+    footnotes = []
+
+    if fairness_ratio > 1:  # backwards compatibility for limits greater than 1
+        fairness_ratio = (fairness_ratio - 1) / fairness_ratio  # (1.25 - 1 ) / 1.25 = 0.25/1.25 = 0.2
+
+    if fairness_ratio <= 0 or fairness_ratio >= 0.5:
+        raise ValueError("Fairness ratio must be between 0 and 0.5")
```

Comment:
I need to think on this one, as the ratio we are looking at is between group 1 and 2, so a ratio of 100 to 1 is just 0.01 which is a 0.495 so technically possible. 

It just means my error warning is messed up since a ratio of .2 is 20% less than the larger of the two metrics. but if you are 100% less than the larger metric, you are zero, which doesn't make sense.

Reply:
This is now setup to allow larger numbers, still arbitrarily stopping at 100, which gives a 2x and 3x change in the metric